index.js

```diff
@@ -79,6 +79,14 @@ class Client {
     const socket = net.connect(this.port, this.hostname)
     socket.setTimeout(options.timeout || this.timeout)
 
+    socket.on('timeout', () => {
+      socket.end()
+      const err = new Error('Request timed out')
```

Comment:
Might be nice to include the `hostname` in the error message so it's easy to tell what service the error is coming from.